assets/app/views/editor/edit-file.js

```diff
@@ -22,55 +23,139 @@ var EditorView = Backbone.View.extend({
 
     this.editors = {};
     this.path = opts.path;
     this.isNewPage = opts.isNewPage || false;
+    this.settingsFields = this.extendSettingFields(opts.settingsFields);
 
     this.model.on('github:commit:success', this.saveSuccess.bind(this));
     this.model.on('github:commit:error', this.saveFailure.bind(this));
 
-    if (!this.isNewPage) {
+    this.doc = this.initializeDocument({
+      fileExt: fileExt,
+      isNewPage: this.isNewPage
+    });
+
+    html.settingsDisplayStyle = this.getSettingsDisplayStyle(this.doc);
+
+    this.$el.html(this.template(html));
+    this.initializeSettingsEditor(this.doc);
+    this.initializeContentEditor(this.doc, fileExt);
+
+    io.socket.get('/v0/site/lock', { file: file }, function(data) {
+
+      // Store the socket ID for future reference
+      self.socket = data.id;
+
+      // Apply the lock
+      self.lockContent.bind(self);
+
+      // On any change events (others open or leave the page), reapply the lock
+      io.socket.on('change', self.lockContent.bind(self));
+
+      // If the user navigates away from the page, remove the lock
+      // The server will do this automatically if the socket session breaks
+      federalist.once('route', function() {
+        $('.alert-container').html('');
+        io.socket.get('/v0/site/unlock', { file: file });
+      });
+
+    });
+
+    return this;
+  },
+  initializeDocument: function (opts) {
+    var fileExt = opts.fileExt,
+        content, doc, raw;
+
+    if (!opts.isNewPage) {
       raw = decodeB64(this.model.attributes.json.content);
       content = this.cleanContent(raw);
-      this.doc = new Document({
-        fileExt: this.model.get('file').split('.')[1],
+      doc = new Document({
+        fileExt: fileExt,
         content: content
       });
     }
     else {
-      this.doc = new Document({
+      doc = new Document({
         fileExt: 'md',
         content: ['---', this.model.getDefaults(), '---', '\n'].join('\n')
       });
     }
 
-    this.$el.html(this.template({ fileName: this.model.get('file') }));
+    return doc;
+  },
+  initializeSettingsEditor: function (doc) {
+    var settings, settingsEditorEl;
 
-    settingsEditorEl = this.$('[data-target=metadata]')[0];
+    settingsEditorEl = this.$('[data-target=settings]')[0];
     this.editors.settings = CodeMirror(settingsEditorEl, {
       lineNumbers: true,
       mode: "yaml",
       tabSize: 2
     });
 
-    if (this.doc.frontMatter) {
-      this.editors.settings.doc.setValue(this.doc.frontMatter);
+    if (doc.frontMatter || doc.frontMatter === '') {
+      this.settings = this.parseSettings(doc);
+      this.editors.settings.doc.setValue(this.settings.remaining);
     }
+  },
+  extendSettingFields: function (fields) {
+    var f = _.extend({
+      title: {
+        type: 'text'
+      },
+      layout: {
+        type: 'select',
+        options: this.model.getLayouts()
+      },
+      author: {
+        type: 'text'
+      },
+      published: {
```

Comment:
[`published`](http://jekyllrb.com/docs/frontmatter/#predefined-global-variables) is actually a reserved attribute in Jekyll. It's a boolean for whether a post or page should render. [`date`](http://jekyllrb.com/docs/frontmatter/#predefined-variables-for-posts) is the key for the publish date for posts, and while it wouldn't do anything by default for pages, we could still use it.